CMakeLists.txt

```diff
@@ -221,6 +221,8 @@ else()
   message(FATAL_ERROR "Invalid polling method")
 endif()
 
+list(APPEND sources ${CMAKE_CURRENT_SOURCE_DIR}/external/sha1/sha1.c ${CMAKE_CURRENT_SOURCE_DIR}/external/sha1/sha1.h)
```

Comment:
as ws_transport is not depending on anything at the moment I'm not sure we need a flag for it.
Can it always be enabled? do you see a problem with it?

Reply:
I think it's fine to have it always enabled given there's no external dependency - it should be DRAFT though, but we can fix that later

Reply:
Mh actually I don't like adding new functionality that can be separated out easily (which is true here) unconditionally. I think that there's already too much unconditionally included. I would tend to prefer to make even more optional, so that it is possible to have a minimal libzmq.